rpc/lib/server/http_server.go

```diff
@@ -54,6 +68,9 @@ func StartHTTPAndTLSServer(listenAddr string, handler http.Handler, certFile, ke
 	if err != nil {
 		return nil, errors.Errorf("Failed to listen on %v: %v", listenAddr, err)
 	}
+	if config.MaxOpenConnections > 0 {
+		listener = netutil.LimitListener(listener, config.MaxOpenConnections)
```

Comment:
Beautiful! What is the behaviour of LimitListener, will it just block on the socket until a connection gets freed up? Is there an attack vector for resource exhaustion other than ports/fds when we use this?

Reply:
> will it just block on the socket until a connection gets freed up

yes

> Is there an attack vector for resource exhaustion other than ports/fds when we use this?

there is still no rate limiting. so one client can occupy all available slots